src/apirequest.ts

```diff
@@ -78,48 +78,29 @@ export function createAPIRequest<T>(
 }
 
 async function createAPIRequestAsync<T>(parameters: APIRequestParams) {
-  const options = Object.assign({}, parameters.options);
-
-  // Create a new params object so it can no longer be modified from outside
-  // code Also support global and per-client params, but allow them to be
-  // overriden per-request
-  const globalParams =
-    parameters.context.google &&
-    parameters.context.google._options &&
-    parameters.context.google._options.params
-      ? parameters.context.google._options.params
-      : {};
-  const clientParams =
-    parameters.context &&
-    parameters.context._options &&
-    parameters.context._options.params
-      ? parameters.context._options.params
-      : {};
-  const params = Object.assign(
+  // Combine the GaxiosOptions options passed with this specific
+  // API call witht the global options configured at the API Context
+  // level, or at the global level.
+  const options = extend(
+    true,
+    {},
+    parameters.context.google?._options || {},
+    parameters.context._options || {},
+    parameters.options
+  );
+
+  const params = extend(
+    true,
     {}, // New base object
-    globalParams, // Global params
-    clientParams, // Per-client params
+    options.params, // Combined global/per-api params
     parameters.params // API call params
   );
 
   // Check for user specified user agents at all levels of config
-  const userAgentGlobal =
-    parameters.context.google &&
-    parameters.context.google._options &&
-    parameters.context.google._options.userAgentDirectives
-      ? parameters.context.google._options.userAgentDirectives
-      : [];
-  const userAgentClient =
-    parameters.context &&
-    parameters.context._options &&
-    parameters.context._options.userAgentDirectives
-      ? parameters.context._options.userAgentDirectives
-      : [];
   const userAgentDirectives: UserAgentDirective[] =
     Object.assign(
       [],
-      userAgentGlobal,
-      userAgentClient,
+      options.userAgentDirectives,
       parameters.options.userAgentDirectives
```

Comment:
This is unclear, I think `parameters.options.userAgentDirectives` is not needed here since the whole content of `parameters.options` is now in `options`, and if it had `userAgentDirectives`, it will be in `options` already?

Reply:
You are totally right.  I just deleted this code, and oddly all of it works (both local tests and linked with googleapis).